### DISCUSSION TITLE
Tailwindcss-language-server unable to find root directory in my vue project

### DISCUSSION THREAD

**Discussion by dre-bk201** [luckasRanarison/tailwind-tools.nvim#66] [Q&A]
Is there a reason why the default `root_dir` pattern for the tailwindcss-lsp was rewritten. Because the [custom one](https://github.com/luckasRanarison/tailwind-tools.nvim/blob/3703e162d819a04b128902c508276553dae0dfb8/lua/tailwind-tools/lsp.lua#L139) is unable to find the root of my project.
```lua
 lspconfig.tailwindcss.setup({
    capabilities = capabilities,
    on_attach = opts.on_attach,
    filetypes = filetypes.get_all(),
    init_options = {
      userLanguages = filetypes.get_server_map(),
    },
    settings = {
      tailwindCSS = opts.settings,
      includeLanguages = filetypes.get_server_map(),
    },
   -- Custom pattern
    root_dir = lspconfig.util.root_pattern(
      "tailwind.config.{js,cjs,mjs,ts}",
      "assets/tailwind.config.{js,cjs,mjs,ts}",
      "theme/static_src/tailwind.config.{js,cjs,mjs,ts}"
    ),
  })
```

After making the changes to the default `root_dir` in [lspconfig](https://github.com/neovim/nvim-lspconfig/blob/ae8a01bb40848490548e5d07b97ed972ed43c2f9/lua/lspconfig/configs/tailwindcss.lua#L102), the plugin started working
```lua
  lspconfig.tailwindcss.setup({
    capabilities = capabilities,
    on_attach = opts.on_attach,
    filetypes = filetypes.get_all(),
    init_options = {
      userLanguages = filetypes.get_server_map(),
    },
    settings = {
      tailwindCSS = opts.settings,
      includeLanguages = filetypes.get_server_map(),
    },
   -- Default working pattern
    root_dir = function(fname)
      return lspconfig.util.root_pattern(
        'tailwind.config.js',
        'tailwind.config.cjs',
        'tailwind.config.mjs',
        'tailwind.config.ts',
        'postcss.config.js',
        'postcss.config.cjs',
        'postcss.config.mjs',
        'postcss.config.ts'
      )(fname) or vim.fs.dirname(vim.fs.find('package.json', { path = fname, upward = true })[1]) or vim.fs.dirname(
        vim.fs.find('node_modules', { path = fname, upward = true })[1]
      ) or vim.fs.dirname(vim.fs.find('.git', { path = fname, upward = true })[1])
    end
  })
```

**Comment by luckasRanarison** [comment]
I just wanted to make it only start on a real tailwind project and not just any node project. Could you show a tree view of your project directory?

**Reply by dre-bk201** [reply]
![{2C5BF682-4F8B-43BF-A10C-E6F970C7EF9F}](https://github.com/user-attachments/assets/00780588-33cf-4f82-8e0f-58694d543891)

**Reply by luckasRanarison** [reply]
Uhm, that's weird. It should start because of the tailwind.config.js file

**Reply by dre-bk201** [reply]
I have no idea either, I just tried creating a fresh project to test with `bun create vite@latest` and choosing the following options:
![{6EA68D25-10E7-4CFE-9124-BAE22977EB4A}](https://github.com/user-attachments/assets/b64a1e0b-d0c0-40b9-a0e5-8f5b8d49dbdd)

Also using windows if that matters
